Resources/Prototypes/Entities/Structures/Power/Generation/Tesla/energyball.yml

```
@@ -0,0 +1,115 @@
+- type: entity
+  id: BaseEnergyBall
+  abstract: true
+  components:
+  - type: Clickable
+  - type: Physics
+    bodyType: Dynamic
+    bodyStatus: InAir
+  - type: CanMoveInAir
+  - type: ChasingWalk
+    minSpeed: 1
+    maxSpeed: 3
+    chasingComponent: LightningTarget
+  - type: AmbientSound
+    volume: 3
+    range: 15
+    sound:
+      path: /Audio/Effects/tesla.ogg
+  - type: PointLight
+    enabled: true
+    radius: 5
+    color: "#4080FF"
+  - type: Appearance
+  - type: LightningArcShooter
+    arcDepth: 2
+    maxLightningArc: 1
+    shootMinInterval: 4
+    shootMaxInterval: 10
+    shootRange: 3
+    lightningPrototype: ChargedLightning
```

```suggestion
    lightningPrototype: Lightning
```
This is why the lightning is having some issues with sprites.

---

Well, I just needed some red dangerous high damage lightning bolts. Well, the fact that the beam system draws them incorrectly, I think, is not a problem of tesla, but a problem of the beam system. It should be possible to write any type of lightning here, and it MUST work, otherwise why this field is needed at all?